### PR TITLE
[Transform] enhance the output of preview to return full destination index details

### PR DESCRIPTION
**Pull request by hendrikmuhs**
changes the output format of preview regarding deduced mappings and enhances it to return all the details about auto-index creation. This allows the user to customize the index creation. Using HLRC you can create a index request from the output of the response. Long term this allows us to add UI support, see https://github.com/elastic/kibana/issues/57059 and/or improve setting defaults.

The output of the preview itself is **not** affected. The UI does not use/display mappings at the moment (CC @walterra )

Breaking:
- supported: 
  - mixed clusters internal communication, all versions
  - HLRC `>=7.7` <-> `<7.7`
- not supported: 
  - HRLC `<7.7`  <-> `>=7.7` (mappings are parsed as empty)

New format:

```
{ 
    "preview": [
       ...
    ],
    "generated_dest_index": {
        "mappings" : {
        ...
        },
        "settings": {
            "index" : {
                "number_of_shards" : 1,
                "auto_expand_replicas" : "0-1",
                ...
            }
        },
        "aliases": {}
    }
}
```

**Please raise concerns about the naming: `generated_dest_index`.**

Old format:

```
{ 
    "preview": [
       ...
    ],
    "mappings" : {
       ...
    }
}
```

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformIndex.java [hunk 1]
```diff
@@ -10,99 +10,135 @@
...
-            builder.field(TYPE, fieldType);
+        Settings settings = createSettings();
 
-            builder.endObject();
-        }
-        builder.endObject(); // PROPERTIES
-        return builder;
+        // transform does not create aliases, however the user might customize this in future
+        Set<Alias> aliases = null;
+        logger.info("mapping {} setting {}", indexMappings, settings);
```
Comment (benwtrent) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformIndex.java]: This feels like `debug` to me.

**Review by benwtrent — APPROVED** [review]
love this change. 

One comment about a logging level, but looks good.

Tricky thing with HLRC, all versions need to be forward-compatible with minor versions in the same major. 

This change should be OK as the HLRC continues to work as a whole.

**Comment by walterra** [comment]
Just caught up with emails and missed the CC unfortunately: We use mappings in the UI to infer field types for the preview tables. I'll come up with a fix this afternoon!

**Comment by walterra** [comment]
Fixed in Kibana for `master/7.x`: https://github.com/elastic/kibana/pull/60609

**Pull request merged**